src/compiler/checker.ts

```diff
@@ -17068,6 +17069,13 @@ namespace ts {
                         return !tp.isThisType && isPartOfTypeNode(node) && maybeTypeParameterReference(node) &&
                             getTypeFromTypeNodeWorker(node as TypeNode) === tp; // use worker because we're looking for === equality
                     case SyntaxKind.TypeQuery:
+                        const entityName = (node as TypeQueryNode).exprName;
+                        const firstIdentifier = getFirstIdentifier(entityName);
+                        const firstIdentifierSymbol = getResolvedSymbol(firstIdentifier);
+                        if (firstIdentifierSymbol.declarations) {
+                            return !!forEach(firstIdentifierSymbol.declarations, idDecl => isNodeDescendantOf(idDecl, tp.symbol.declarations![0].parent)) ||

```

Comment:
Use the `some` function instead of `forEach`. Also, to get the declaration of `tp` use `getDeclarationOfKind<TypeParameterDeclaration>(tp.symbol, SyntaxKind.TypeParameter)!`. Although very unlikely, it is possible for there to be other declarations in the array, for example if the same name is used for a type parameter and a regular parameter.